### PR TITLE
Feat: Discord connector

### PR DESCRIPTION
**Pull request by tabossert**
This connector takes discord channel ids, a token and other options to pull conversation history for a channel and store it as a text file that is then processed by unstructured into expected output.

Related to: https://github.com/Unstructured-IO/unstructured/issues/253

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -118,14 +122,15 @@ pip-compile:
...
 	cp requirements/build.txt docs/requirements.txt
-	pip-compile --upgrade --extra=s3        --output-file=requirements/ingest-s3.txt        requirements/base.txt setup.py
-	pip-compile --upgrade --extra=azure     --output-file=requirements/ingest-azure.txt     requirements/base.txt setup.py
-	pip-compile --upgrade --extra=reddit    --output-file=requirements/ingest-reddit.txt    requirements/base.txt setup.py
-	pip-compile --upgrade --extra=github    --output-file=requirements/ingest-github.txt    requirements/base.txt setup.py
-	pip-compile --upgrade --extra=gitlab    --output-file=requirements/ingest-gitlab.txt    requirements/base.txt setup.py
-	pip-compile --upgrade --extra=slack     --output-file=requirements/ingest-slack.txt     requirements/base.txt setup.py
-	pip-compile --upgrade --extra=wikipedia --output-file=requirements/ingest-wikipedia.txt requirements/base.txt setup.py
-	pip-compile --upgrade --extra=google-drive --output-file=requirements/ingest-google-drive.txt  requirements/base.txt setup.py
+	pip-compile --upgrade --extra=s3        	--output-file=requirements/ingest-s3.txt        	requirements/base.txt setup.py
```
Comment (MthwRobinson) [inline Makefile:125]: Are these formatting changes intentional? Looks like there's extra space between the existing flags.

File: discord-test/1100149908494876775.txt [hunk 2]
```diff
@@ -0,0 +1,2 @@
+😀
+https://tenor.com/view/test-homer-simpson-mouse-rat-lab-gif-19273011
```
Comment (MthwRobinson) [inline discord-test/1100149908494876775.txt:2]: 🐭

File: scripts/ingest-test-fixtures-update.sh [hunk 3]
```diff
@@ -47,4 +47,5 @@ docker run --rm -v "$SCRIPT_DIR"/../unstructured:/root/unstructured -v \
                ./test_unstructured_ingest/test-ingest-biomed-api.sh &&
                ./test_unstructured_ingest/test-ingest-biomed-path.sh &&
                ./test_unstructured_ingest/test-ingest-s3.sh &&
+               ./test_unstructured_ingest/test-ingest-slack.sh &&
```
Comment (MthwRobinson) [inline scripts/ingest-test-fixtures-update.sh:50]: Should one of these be `test-ingest-discord.sh`?

**Review by MthwRobinson — APPROVED** [review]
Couple small inline comments, otherwise LGTM from my end if the new ingest test is passing for this one.

File: Makefile (hunk 1 above)
Comment (tabossert) [reply Makefile]: Fixed

**Pull request merged**